// _ng-reusable-auth/index.js
import { Injectable, NgModule } from '@angular/core';
import { NotifyModule, NotifyService } from '@agiliumlabs/ng2-notify';
import { BrowserModule } from '@angular/platform-browser';

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var ReusableService = (function () {
    function ReusableService() {
    }
    /**
     * @return {?}
     */
    ReusableService.prototype.getSomething = function () {
        return 'Something from reusable service';
    };
    return ReusableService;
}());
ReusableService = __decorate([
    Injectable()
], ReusableService);

var __decorate$1 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (undefined && undefined.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var SignInSharedComponent = (function () {
    /**
     * @param {?} _notify
     * @param {?} _reusable
     */
    function SignInSharedComponent(_notify, _reusable) {
        this._notify = _notify;
        this._reusable = _reusable;
    }
    /**
     * @return {?}
     */
    SignInSharedComponent.prototype.callReusableNotify = function () {
        this._notify.notifySuccess('sign-in', 'Reusable module in reusable auth work!');
    };
    /**
     * @return {?}
     */
    SignInSharedComponent.prototype.callReusableService = function () {
        var /** @type {?} */ someMessage = this._reusable.getSomething();
        someMessage = "Reusable service return data: " + someMessage;
        alert(someMessage);
    };
    return SignInSharedComponent;
}());
SignInSharedComponent = __decorate$1([
    Injectable(),
    __metadata("design:paramtypes", [NotifyService,
        ReusableService])
], SignInSharedComponent);

var __decorate$2 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata$1 = (undefined && undefined.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var SignUpSharedComponent = (function () {
    /**
     * @param {?} _notify
     */
    function SignUpSharedComponent(_notify) {
        this._notify = _notify;
    }
    /**
     * @return {?}
     */
    SignUpSharedComponent.prototype.callReusableNotify = function () {
        this._notify.notifySuccess('sign-up', 'Sign up notification');
    };
    return SignUpSharedComponent;
}());
SignUpSharedComponent = __decorate$2([
    Injectable(),
    __metadata$1("design:paramtypes", [NotifyService])
], SignUpSharedComponent);

var __decorate$3 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var ReusableModule = (function () {
    function ReusableModule() {
    }
    return ReusableModule;
}());
ReusableModule = __decorate$3([
    NgModule({
        imports: [
            BrowserModule,
            NotifyModule
        ],
        declarations: [],
        exports: [
            NotifyModule
        ],
        providers: [
            ReusableService
        ]
    })
], ReusableModule);

export { ReusableService, SignInSharedComponent, SignUpSharedComponent, ReusableModule };
