// _ng-reusable-auth/ng-reusable-auth.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@agiliumlabs/ng2-notify'), require('@angular/platform-browser')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core', '@agiliumlabs/ng2-notify', '@angular/platform-browser'], factory) :
	(factory((global['ng-reusable-auth'] = global['ng-reusable-auth'] || {}),global._angular_core,global._agiliumlabs_ng2Notify,global._angular_platformBrowser));
}(this, (function (exports,_angular_core,_agiliumlabs_ng2Notify,_angular_platformBrowser) { 'use strict';

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
exports.ReusableService = (function () {
    function ReusableService() {
    }
    /**
     * @return {?}
     */
    ReusableService.prototype.getSomething = function () {
        return 'Something from reusable service';
    };
    return ReusableService;
}());
exports.ReusableService = __decorate([
    _angular_core.Injectable()
], exports.ReusableService);

var __decorate$1 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (undefined && undefined.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
exports.SignInSharedComponent = (function () {
    /**
     * @param {?} _notify
     * @param {?} _reusable
     */
    function SignInSharedComponent(_notify, _reusable) {
        this._notify = _notify;
        this._reusable = _reusable;
    }
    /**
     * @return {?}
     */
    SignInSharedComponent.prototype.callReusableNotify = function () {
        this._notify.notifySuccess('sign-in', 'Reusable module in reusable auth work!');
    };
    /**
     * @return {?}
     */
    SignInSharedComponent.prototype.callReusableService = function () {
        var /** @type {?} */ someMessage = this._reusable.getSomething();
        someMessage = "Reusable service return data: " + someMessage;
        alert(someMessage);
    };
    return SignInSharedComponent;
}());
exports.SignInSharedComponent = __decorate$1([
    _angular_core.Injectable(),
    __metadata("design:paramtypes", [_agiliumlabs_ng2Notify.NotifyService,
        exports.ReusableService])
], exports.SignInSharedComponent);

var __decorate$2 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata$1 = (undefined && undefined.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
exports.SignUpSharedComponent = (function () {
    /**
     * @param {?} _notify
     */
    function SignUpSharedComponent(_notify) {
        this._notify = _notify;
    }
    /**
     * @return {?}
     */
    SignUpSharedComponent.prototype.callReusableNotify = function () {
        this._notify.notifySuccess('sign-up', 'Sign up notification');
    };
    return SignUpSharedComponent;
}());
exports.SignUpSharedComponent = __decorate$2([
    _angular_core.Injectable(),
    __metadata$1("design:paramtypes", [_agiliumlabs_ng2Notify.NotifyService])
], exports.SignUpSharedComponent);

var __decorate$3 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
exports.ReusableModule = (function () {
    function ReusableModule() {
    }
    return ReusableModule;
}());
exports.ReusableModule = __decorate$3([
    _angular_core.NgModule({
        imports: [
            _angular_platformBrowser.BrowserModule,
            _agiliumlabs_ng2Notify.NotifyModule
        ],
        declarations: [],
        exports: [
            _agiliumlabs_ng2Notify.NotifyModule
        ],
        providers: [
            exports.ReusableService
        ]
    })
], exports.ReusableModule);

Object.defineProperty(exports, '__esModule', { value: true });

})));
